react-tether/index.d.ts

```diff
@@ -0,0 +1,25 @@
+// Type definitions for react-tether 0.5
+// Project: https://github.com/souporserious/react-tether
+// Definitions by: Ryan Price <https://github.com/ryprice>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.1
+
+export default TetherComponent;
+export as namespace ReactTether;
+
+import * as React from 'react';
+import * as Tether from 'tether';
+
+declare class TetherComponent extends React.Component<ReactTether.TetherComponentProps, null> { }
+
+declare namespace ReactTether {
+    export interface TetherComponentProps extends React.Props<TetherComponent>, Tether.ITetherOptions {
+        renderElementTag?: string;
+        renderElementTo?: string | {appendChild: (element: HTMLElement) => void};
+        id?: string;
+        className?: string;
+        style?: React.CSSProperties;
+        onUpdate?: () => void;
+        onRepositioned?: () => void;
+    }
```

Comment:
Not really sure what the valid args are for onUpdate and onRepositioned because it's somewhat obscure from the react-tether code, but the linter wasn't pleased with Function.